File: scripts/generate-global-drug-seed.mjs
#!/usr/bin/env node
/** Regenerate multi-region finished-drug seed + CN brand aliases.
 * Prefers scripts/_generate_global_drug_seed.py if present; otherwise validates existing JSON.
 */
import { existsSync, readFileSync } from "node:fs";
import { dirname, join } from "node:path";
import { fileURLToPath } from "node:url";
import { spawnSync } from "node:child_process";

const root = join(dirname(fileURLToPath(import.meta.url)), "..");
const py = join(root, "scripts/_generate_global_drug_seed.py");
const cn = join(root, "data/incoming/open/cn-brand-aliases.json");
const glob = join(root, "data/incoming/ndc/global-products-seed.json");

if (existsSync(py)) {
  const r = spawnSync("python3", [py], { cwd: root, stdio: "inherit" });
  process.exit(r.status || 0);
}

if (!existsSync(cn) || !existsSync(glob)) {
  console.error("Missing seed JSON. Expected:", cn, "and", glob);
  process.exit(1);
}
const a = JSON.parse(readFileSync(cn, "utf8"));
const g = JSON.parse(readFileSync(glob, "utf8"));
const products = Array.isArray(g) ? g : g.products || [];
const aliases = Array.isArray(a) ? a : a.aliases || [];
const by = {};
for (const p of products) for (const t of p.countryTags || []) by[t] = (by[t] || 0) + 1;
console.log("CN brand aliases:", aliases.length);
console.log("Global products:", products.length, by);
console.log("Seeds already present — edit JSON or add scripts/_generate_global_drug_seed.py to rebuild.");

File: scripts/_generate_global_drug_seed.py
#!/usr/bin/env python3
"""Regenerate / validate multi-region finished-drug seed + CN brand aliases.
Identity layer only — no monograph text. Prefer extending JSON in-place;
this script reports counts and ensures wrapper metadata is consistent.
"""
from __future__ import annotations
import json
from datetime import datetime, timezone
from pathlib import Path

ROOT = Path(__file__).resolve().parents[1]
CN = ROOT / "data/incoming/open/cn-brand-aliases.json"
GLOB = ROOT / "data/incoming/ndc/global-products-seed.json"

def load(path: Path):
    data = json.loads(path.read_text(encoding="utf-8"))
    return data

def main():
    now = datetime.now(timezone.utc).strftime("%Y-%m-%dT%H:%M:%S.000Z")
    cn = load(CN)
    aliases = cn.get("aliases") if isinstance(cn, dict) else cn
    g = load(GLOB)
    products = g.get("products") if isinstance(g, dict) else g
    by = {}
    for p in products:
        for t in p.get("countryTags") or []:
            by[t] = by.get(t, 0) + 1
    cn_out = {
        "generatedAt": now,
        "count": len(aliases),
        "note": "Curated open Chinese brand/trade ↔ INN/UNII synonym map (identity only)",
        "aliases": aliases,
    }
    g_out = {
        "generatedAt": now,
        "count": len(products),
        "note": "Multi-region finished-drug identity seed (WHO/EMA/MHRA/PMDA/CN curated). No monograph text.",
        "products": products,
    }
    CN.write_text(json.dumps(cn_out, ensure_ascii=False, indent=2) + "\n", encoding="utf-8")
    GLOB.write_text(json.dumps(g_out, ensure_ascii=False) + "\n", encoding="utf-8")
    print("CN brand aliases:", len(aliases))
    print("Global products:", len(products), by)
    print("Wrote", CN, "and", GLOB)

if __name__ == "__main__":
    main()
